### ISSUE TITLE
4.19: GLOB_BRACE is not portable

### ISSUE THREAD

**Issue by kanavin** [rpm-software-management/rpm#2844]
When building 4.19.1 with musl C library, the following error happens:
```
/home/pokybuild/yocto-worker/musl-qemux86-64/build/build/tmp/work/core2-64-poky-linux-musl/rpm/4.19.1/git/rpmio/rpmglob.c: In function 'rpmGlobPath':
| /home/pokybuild/yocto-worker/musl-qemux86-64/build/build/tmp/work/core2-64-poky-linux-musl/rpm/4.19.1/git/rpmio/rpmglob.c:84:15: error: 'GLOB_BRACE' undeclared (first use in this function); did you mean 'GLOB_NOSPACE'?
|    84 |     gflags |= GLOB_BRACE;
|       |               ^~~~~~~~~~
|       |               GLOB_NOSPACE
```

I appreciate that fixing this isn't easy (unless one wants to break all usages of braces in calls to the function).

Full log (it's large, and doesn't contain anything else that would be relevant):
https://autobuilder.yoctoproject.org/typhoon/#/builders/45/builds/8466/steps/11/logs/stdio

**Comment by dmnks** [comment]
Yup, we recently refactored quite a bit of code related to glob handling, and mostly went with the GNU `glob(3)` implementation, meaning that at least `GLOB_BRACE` is indeed used now. Previously, we had our own implementation of brace expansion, IIRC.

I'll have a closer look but it's possible this could actually be fixed upstream. Thanks for the report!

**Comment by kanavin** [comment]
FWIW, I think it's most easily fixable by relying on gnulib implementation?
https://github.com/coreutils/gnulib/blob/master/lib/glob.c
